src/test/fakeSuite/extension.test.ts

```diff
@@ -1,286 +1,330 @@
+    // (see comment in parser.ts - parseLineAsTool), so again, we need to find the toolset that is referenced in the makefile.
+    // TODO: mock various scenarios of VS environments without depending on what is installed.
+    // TODO: adapt the makefile on mac/linux/mingw and add new tests in this suite
+    // to parse the dry-run logs obtained on those platforms.
+    let systemPlatform: string;
+    if (process.platform === "win32") {
+        systemPlatform = (process.env.MSYSTEM === undefined) ? "win32" : process.env.MSYSTEM;
+    } else {
+        systemPlatform = process.platform;
+    }
+
+    if (systemPlatform === "win32") {
+        test('Interesting small makefile - windows', async() => {
+            // Settings reset from the previous test run.
+            extension.getState().reset(false);
+            await configuration.initFromStateAndSettings();
+
+            // We define extension log here as opposed to in the fake repro .vscode/settings.json
+            // because the logging produced at the first project load has too few important data to verify and much variations
+            // that are not worth to be processed when comparing with a baseline.
+            // Example: when running a test after incomplete debugging or after loading the fake repro project independently of the testing framework,
+            // which leaves the workspace state not clean, resulting in a different extension output log
+            // than without debugging/loading the project before.
+            // If we define extension log here instead of .vscode/settings.json, we also have to clean it up
+            // because at project load time, there is no makefile log identified and no file is deleted on activation.
+            let extensionLogPath: string = path.join(vscode.workspace.rootPath || "./", ".vscode/Makefile.out");
+            if (util.checkFileExistsSync(extensionLogPath)) {
+                util.deleteFileSync(extensionLogPath);
+            }
+            configuration.setExtensionLog(extensionLogPath);
+
+            // Run a preconfigure script to include our tests "Program Files" path so that we always find a cl.exe
+            // from this extension repository instead of a real VS installation that happens to be in the path.
+            configuration.setPreConfigureScript(path.join(vscode.workspace.rootPath || "./", ".vscode/preconfigure.bat"));
+            await make.preConfigure(make.TriggeredBy.tests);
+
+            configuration.prepareConfigurationsQuickPick();
+            configuration.setConfigurationByName("InterestingSmallMakefile_windows_configDebug");
+            const retc: number = await make.cleanConfigure(make.TriggeredBy.tests, true);
+
+            configuration.setBuildBeforeLaunch(false);
+            const launchConfigurations: string[] = ["bin\\InterestingSmallMakefile\\ARC H3\\Debug\\main.exe(str3a,str3b,str3c)",

```

Comment:
This one is not checked in but the others are. Do we need these exe's checked in, or can we .gitignore them?

Reply:
We need them checked in. Depending on finding them or not, we parse extra launch targets (the ones which are directly invoked, with or without parameters, in the makefile). When these binaries are missing, we still detect one launch target for each, based on what the linker produces. We have information from the -out or -o switches of any linker that we support. But when the makefile invokes one such built binary with particular arguments, then we need to parse that invocation from the dryrun log. And we need to check if it exists on disk to be able to reject other lines where there isn't such built binary invocation but something else which happens to have the name of that binary target in the middle. It's a limitation of the extension not being able to smartly figure out a path versus non path to be able to reject some matches.

Reply:
So are you going to check in `bin\\InterestingSmallMakefile\\ARC H3\\Debug\\main.exe`?

Reply:
It is already. You're saying you see it is not?
Since we're talking about these exe's being checked in, maybe I should un-checkin one of them and update baselines. When the binary is not there, we can't parse safely the invocations of that binary with arguments. But for a future version.

Reply:
It wasn't in this PR.  If it's already checked in, then everything should be fine.